haystack/nodes/question_generator/question_generator.py

```diff
@@ -60,6 +60,11 @@ def __init__(
         :param model_version: The version of model to use from the HuggingFace model hub. Can be tag name, branch name, or commit hash.
         :param use_gpu: Whether to use GPU or the CPU. Falls back on CPU if no GPU is available.
         :param batch_size: Number of documents to process at a time.
+        :param num_queries_per_doc: Number of questions to generate per document. However, this is actually a number
+                                    of question to generate per split in the document where the `split_length` determines
+                                    the length of the split and the `split_overlap` determines the overlap between splits.
+                                    Therefore, this parameter is multiplied by the resulting number of splits to get the
+                                    total number of questions generated per document. This value is capped at 3.
```

Comment:
Hey, @vblagoje is there a reason why we want to cap this value? It might just be sufficient to warn the user that increasing this value will slow down the generation.

Reply:
It will not only slow down, but it will generate horrible questions. Think of it this way - you have 100 tokens, and you ask a model to generate x questions on those 100 tokens where x is potentially a large number. I've played with it yesterday, and after 3 questions the quality becomes abysmal.

Reply:
And not only that but we'll deviate even more from the original goal of num_queries_per_doc per doc. This value should perhaps be hard coded to 1 :-)

Reply:
Yes exactly!!!

Reply:
So if input a long document (e.g. 1000+ words), does that document get truncated and only the first `model_max_length` tokens get considered?

Reply:
No, sorry, it is a number of queries per split. So if the document has x splits and num_queries_per_doc is y, we'll have x*y questions per document

Reply:
Ahh okay, I understand. Thanks for the explanation! And I agree capping to something like 3 should be fine.